students/darrell_h/lesson_6/mailroom_with_exceptions.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/local/bin/python3
+"""This module does blah blah."""
+
+
+import datetime
+import os  # this is required to write files to ~/Downloads. os.path.expanduser
+
+
+##############
+#    DATA    #
+##############
+
+d = {'jack': [100, 200, 300, 400],
+     'mary': [3000, 5000],
+     'frank': [29.50, 31],
+     'jane': [3000, 5000],
+     'scrouge': [1, 2, 3],
+     'bob': [60000, 70000, 7668, 4]}
+
+
+#####################
+#  HELPER FUNCTIONS #
+#####################
+
+
+def seperator(str):
+    """return line that equals string length ignoring newline."""
+    return "-" * (len(str) - str.count('\n'))
+
+
+def list_donors():
+    print("\nExisting donors\n-" + "\n-".join(d.keys()))
+
+
+def add_donation(donor_name, exsting_donor):
+    while True:
+        donation = input(
+            f"\nEnter a donation amount for {donor_name.capitalize()}: ")
+        if donation.isnumeric() and exsting_donor:
```

Comment:
`exsting_donor` is one letter shy to making it a complete word.  Just use the complete word.